pkg/config/store.go

```diff
@@ -0,0 +1,134 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/configmap"
+)
+
+type cfgKey struct{}
+
+const (
+	// PolicyControllerConfigName is the name of the configmap used to configure
+	// policy-controller.
+	PolicyControllerConfigName = "config-policy-controller"
+
+	// Specifies that if an image is not found to match any policy, it should
+	// be rejected.
+	DenyAll = "deny"
+
+	// Specifies that if an image is not found to match any policy, it should
+	// be allowed.
+	AllowAll = "allow"
+
+	WarnAll = "warn"
+
+	NoMatchPolicyKey = "no-match-policy"
+)
+
+// PolicyControllerConfig controls the behaviour of policy-controller that needs
+// to be more flexible than requiring a controller restart. Some examples are
+// controlling behaviour for what to do if no matching policies are found.
+// Point is that these apply to the whole controller instead of specific CIP
+// policies that apply only to matching images.
+type PolicyControllerConfig struct {
+	// NoMatchPolicy says what do in the case where an image does not match
+	// any policy.
+	NoMatchPolicy string `json:"no-match-policy"`
+}
+
+func NewPolicyControllerConfigFromMap(data map[string]string) (*PolicyControllerConfig, error) {
+	ret := &PolicyControllerConfig{NoMatchPolicy: "deny"}
```

Comment:
The default behavior should be `admit`, right ? At least, the configmap sets the default to `admit`.

Reply:
Yes, that's the default behaviour out of the box. But, in case of errors, I believe doing the secure thing by default is correct thing to do.

Reply:
We'll need to document this behaviour.